### PR TITLE
tests: rewrite print_hardware() due to nitpicks

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: tests/sha256/main.c [hunk 1]
```diff
@@ -19,35 +19,37 @@ static const size_t num_perf = sizeof(perfsizes) / sizeof(perfsizes[0]);
...
 {
 	int use_hardware = 0;
 
+	/* Inform the user of the general topic... */
+	printf("%s", str);
+
+	/* ... and whether we're using hardware acceleration... */
 #if defined(CPUSUPPORT_X86_SHANI) && defined(CPUSUPPORT_X86_SSSE3)
-	if (cpusupport_x86_shani() && cpusupport_x86_ssse3())
+	if (cpusupport_x86_shani() && cpusupport_x86_ssse3()) {
```
Comment (cperciva) [inline tests/sha256/main.c:33]: This feels unnecessarily complicated.  Why not

```
#if defined(CPUSUPPORT_X86_SHANI) && defined(CPUSUPPORT_X86_SSSE3)
	if (cpusupport_x86_shani() && cpusupport_x86_ssse3()) {
                printf(" using hardware SHANI.\n");
        } else
#endif
#if defined(CPUSUPPORT_X86_SSE2)
	if (cpusupport_x86_sse2()) {
                printf(" using hardware SSE2.\n");
       } else
#endif
...
```

Comment (cperciva) [reply tests/sha256/main.c:33]: I don't think there's any need for `use_hardware` at all in here.

**Comment by gperciva** [comment]
I suppose that my uncomfortableness with `if()` statements spanning preprocessor defines in a weakness.

Updated accordingly.

**Comment by cperciva** [comment]
Normaly I would avoid having an `if` which spans preprocessor conditionals, but in this case it makes everything else so much cleaner I think it's worth it. :-)

**Pull request merged**